### PR TITLE
Shipment callback state update

### LINKED ISSUE #7462: Shipment state indicates 'ready'  and not 'shipped'
## Description
Until the latest release (April 20, 2021),   when an enterprise manager clicked the shipping icon on an order,  the shipment status changed to 'shipped',  this helps the manager keep track of shipped and not shipped orders.  

## Expected Behavior
when the manager selects the small shipping icon on the order listing page,    the shipment status should change to "shipped"

## Actual Behaviour
When the manager clicks the shipping icon on the order listing page,  the shipment icon disappears,  but the shipment state still reads 'ready'  instead of 'shipped'

NOTE -  the shipment confirmation email IS appropriately sent to the customer.

## Steps to Reproduce

1.Place an order in a shop
2.Click the  shipping icon on the orders listing page
3.
4.

## Severity
s2?   if the shipment status doesn't change to 'shipped'  there is no way for an enterprise manager to search their orders listing to tell which orders have shipped and which have not.    No report (that I can see) provides shipping status of orders.
acknowledge order (check) as paid and shipment state is ready
click ship icon and shipment state still says ready

bug-s1: a critical feature is broken: checkout, payments, signup, login
bug-s2: a non-critical feature is broken, no workaround
bug-s3: a feature is broken but there is a workaround
bug-s4: it's annoying, but you can use it
bug-s5: we can live with it, only a few users impacted

https://github.com/openfoodfoundation/openfoodnetwork/wiki/Bug-severity
-->

## Your Environment

* Version used:
* Browser name and version:
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Closes #7462

Brings in a missed change to `Spree::Shipment` from the 2.2 branch. This should have been added along with recent changes to shipment callbacks.

#### What should we test?

Order shipment state should say "shipped" after clicking the "ship" button, and not "ready".

#### Release notes

Changelog Category: Technical changes

### CODE REVIEW

**Review by andrewpbrett — APPROVED** [review]

File: app/models/spree/shipment.rb [hunk 1]
```diff
@@ -331,6 +331,15 @@ def after_ship
...
       send_shipped_email
       touch :shipped_at
+      update_order_shipment_state
+    end
+
+    def update_order_shipment_state
+      new_state = order.updater.update_shipment_state
+      order.update_columns(
+        shipment_state: new_state,
+        updated_at: Time.now,
```
Comment (sauloperez) [inline app/models/spree/shipment.rb:341]: I think it's safe to also use `Time.zone.now`. Rails seems to transparently save it in UTC regardless of the app's timezone therefore, it's a thing less to worry about; always use `Time.zone.now`.

**Review by sauloperez — APPROVED** [review]

File: app/models/spree/shipment.rb (hunk 1 above)
Comment (Matt-Yorkley) [reply app/models/spree/shipment.rb]: Ah, I pulled it straight from Spree! Yeah, we want `Time.zone.now` :100:

**Comment by filipefurtad0** [comment]
Awesome @Matt-Yorkley and @andrewpbrett ,
Thanks for addressing this so quickly.

The shipment state now changes accordingly:

Peek 2021-04-29 10-17

Ready to go.

**Comment by sauloperez** [comment]
oh wow! :heart_eyes: 
Screenshot from 2021-04-29 12-01-10

:clap: :clap:

**Pull request merged**